cli/flox/src/commands/show.rs

```diff
@@ -38,7 +39,21 @@ impl Show {
 
         let (results, exit_status) = if let Some(client) = flox.catalog_client {
             tracing::debug!("using catalog client for show");
-            (client.package_versions(&self.search_term).await?, None)
+            match client.package_versions(&self.search_term).await {
+                Ok(results) => (results, None),
+                // Below, results.is_empty() is used to mean the search_term
+                // didn't match a package.
+                // So translate 404 into an empty vec![].
+                // Once we drop the pkgdb code path, we can clean this up.
+                Err(VersionsError::Versions(e)) if e.status() == 404 => (
```

Comment:
The semantics here are unclear, but the API should not return a 404 for a search which returns no results.  It's a post operation, performing an action, and if it can complete that action successfully, then it should return a success (200) with the results (empty).  IMO, a 404 would be appropriate for a `GET /packages/<attr_path>/` kind of thing.  The point being, if we get a 404 from `/catalog/search`, then it's an exception we should treat as an unexpected error and not silently consider it an empty search result.

Reply:
seconded
404 corresponds to a resource not being found, i'd interpret the resource being "search" here

Reply:
Not sure if I'm misunderstanding, but it seems like you think this code path is for search, but it's for show/`GET /packages/<attr_path>`

Reply:
Ah, got it.  Yes, I was confused on that part, seeing `SearchResults`. Thanks for the clarification.